docs/source/libmav/libmav.py

```diff
@@ -0,0 +1,1046 @@
+    message_id = heartbeat_message.id_for_message('HEARTBEAT')
+    heartbeat_message = message_set.create(message_id)
+        
+    # Create a HEARTBEAT message from its name.       
+    heartbeat_message = message_set.create('HEARTBEAT')  
+    ```
+    
+    Note that the new messages are created with default values.
+    
+    """  
+
+    def __init__(self):
+        """Construct an empty `MessageSet`.
+        
+        ```python
+        # Construct an empty message set
+        message_set = libmav.MessageSet()
+        ```
+        
+        Definitions can be added to the new set using `add_from_xml_file()` or `add_from_xml_string()`.
+
+        Returns:
+            MessageSet: An empty `MessageSet`.
+        """
+        pass
+        
+    def __init__(self, definition_file):
+        """Construct an empty `MessageSet`.
+        
+        Additional definitions can be added to the new set using `add_from_xml_file()` or `add_from_xml_string()`.
+        <!-- what happens if there are clashes/overloads?? -->
+
+        ```python
+        # Construct the common message set
+        message_set = libmav.MessageSet('./mavlink/message_definitions/v1.0/common.xml')
+        
+        # Construct the ardupilotmega message set. 
+        message_set = libmav.MessageSet('./mavlink/message_definitions/v1.0/ardupilotmega.xml')
+        ```
+
+        Args:
+            definition_file (str): Full path to the XML definition file to load.
+
+        Returns:
+            MessageSet: A new `MessageSet` object populated with the definitions in the indicated file.
+        """
+        pass        
+             
+
+    def add_from_xml_file(self, definition_file):
+        """Add definitions from a MAVLink XML definition file.
+        
+        The new definitions are appended to the  to the `MessageSet`.
+        Overloaded definitions are replaced. <!-- is this correct? --> 
+
+        ```python
+        # Construct an empty message set
+        message_set = libmav.MessageSet()
+        
+        # Add the common message set
+        message_set.add_from_xml_file('./mavlink/message_definitions/v1.0/common.xml')
+        ```
+
+        Args: 
+            definition_file (string): Full path to the XML definition file to load.
+        """
+
+    def add_from_xml_string(self, xml_string):
+        """Add XML definitions defined in a string.
+               
+        The new definitions are appended to the `MessageSet`.
+        Overloaded definitions are replaced. <!-- is this correct? --> 
+        
+        The example code below shows the approach.
+
+        ```python
+        message_set.add_from_xml_string('''
+          <mavlink>
+            <messages>
+              <message>
+                ...
+              </message>
+            </messages>
+          </mavlink>''')
+        ```
+
+        Args: 
+            xml_string (str): Definition for XML messages and enums.
+        """
+
+    def create(self, message_name):
```

Comment:
I haven't been consistent with camel case vs snake case. If you set a preference I can update.

Reply:
I don't mind at all, I usually do CamelCase for classes and snake_case for variables

Reply:
Looks like we're not too far off then. Good enough for now.